openassessment/data.py

```diff
@@ -498,6 +498,56 @@ def collect_ora2_data(cls, course_id):
         ]
         return header, rows
 
+    @classmethod
+    def collect_ora2_summary(cls, course_id):
+        """
+        Query database for aggregated ora2 summary data.
+
+        Args:
+            course_id (string) - the course id of the course whose data we would like to return
+
+        Returns:
+            A tuple containing two lists: headers and data.
+
+            headers is a list containing strings corresponding to the column headers of the data.
+            data is a list of lists, where each sub-list corresponds to a row in the table of all the data
+                for this course.
+
+        """
+
+        items = AssessmentWorkflow.objects.filter(course_id=course_id)
+
+        rows = []
+        for aw in items:
+            statuses = aw.status_details()
+            submission_dict = sub_api.get_submission_and_student(aw.submission_uuid)
+
+            row = [
+                aw.item_id,
+                submission_dict['student_item']['student_id'],
+            ] + [
+                'complete' if statuses.get(step) and statuses[step]['complete'] else '' for step in AssessmentWorkflow.STEPS
```

Comment:
Since some steps will never be complete (if they're not present in the assessment workflow, I think we need to indicate this for the row value:
* `complete` if statuses.get(step) and statuses[step]['complete']
* `NA` if not statuses.get(step)
* `''` otherwise.

(And since this is more complex, it's best moved to its own block of code, instead of done in one line.)

Also, need to ensure that the strings used here (`complete`, `NA`) are translatable -- so wrap them in `ugettext`/ `_()` calls.